spec/features/admin/delete_custom_organosation_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'support/confirmation_use_case'
+
+describe 'deleting a custom organisation name' do
+  include_examples 'confirmation use case spy'
+
+  let!(:admin_user) { create(:user, super_admin: true) }
+
+  context 'when visiting the custom organisations page' do
+    before do
+      sign_in_user admin_user
+      visit admin_custom_organisations_path
+      CustomOrganisationName.create(name: 'DummyOrg1')
+      CustomOrganisationName.create(name: 'DummyOrg2')
+    end
+
+    it 'allows the user to delete a custom organisation' do
+      within("tr#data-id-1") do
```

Comment:
You need to do two things here. You need to match the selector with the format of your `id` / `data-id` and you need to pass in the `id` of the `CustomOrganisationName` you are targeting.

I would do something like if you were using `data-id=7`:
```
let!(:custom_org_name) { CustomOrganisationName.create(name: 'DummyOrg1') }

within('tr[data-id="#{custom_org_name.id}"]') do
end
```
or if you are just doing `id=7` the following may work
```
html_selector = "tr#" + custom_org_name.id
within(html_selector) do
end
```